libs/portals/core/src/utils/modules.ts

```diff
@@ -82,3 +82,31 @@ export const arrangeRoutes = async ({
 
   return filteredRoutes
 }
+
+type GetActiveModuleArgs = {
+  userInfo: User
+  modules: PortalModule[]
+  apolloClient: ApolloClient<NormalizedCacheObject>
+  pathname: string
+}
+
+export const getActiveModule = ({
+  pathname,
+  userInfo,
+  modules,
+  apolloClient,
+}: GetActiveModuleArgs) =>
```

Comment:
We should move all of this logic into ModuleProvider. It can use `useLocation` and handle everything in one render. This `useEffect` implementation involves two (pretty large) renders and may even flash the wrong Layout for a ms.